src/explore-education-statistics-common/src/components/form/rhf/validation/RHFCreateErrorHelper.ts

```diff
@@ -0,0 +1,70 @@
+import get from 'lodash/get';
+import {
+  FieldErrors,
+  FieldNamesMarkedBoolean,
+  FieldValues,
+} from 'react-hook-form';
+
+const RHFCreateErrorHelper = <T extends FieldValues>({
+  touched,
+  errors,
+  showErrorsForUntouchedFields = false,
+}: {
+  touched: Partial<Readonly<FieldNamesMarkedBoolean<T>>>;
+  errors: FieldErrors<T>;
+  showErrorsForUntouchedFields?: boolean;
+}) => {
+  const getAllErrors = (
+    errorGroup: FieldValues,
+    keyPrefix?: string,
+  ): { [key: string]: string } => {
+    return Object.entries(errorGroup).reduce((acc, [key, error]) => {
+      const errorKey = keyPrefix ? `${keyPrefix}.${key}` : key;
+
+      const isTouched = get(touched, errorKey, false);
+
+      if (
+        (!showErrorsForUntouchedFields && !isTouched) ||
+        typeof error === 'undefined'
+      ) {
+        return acc;
+      }
+
+      if (error.message) {
+        return {
+          ...acc,
+          [errorKey]: error.message,
+        };
+      }
+
+      return {
+        ...acc,
+        ...getAllErrors(error, errorKey),
+      };
+    }, {});
+  };
+
+  const getError = (name: keyof T | string): string => {
+    const isTouched = get(touched, name, false);
+
+    if (!isTouched) {
+      return '';
+    }
```

Comment:
Given we have a new flag for including errors for touched/untouched errors, would have expected it to also affect the `getError` function too.

As my above comment, we could add a new parameter to this function to toggle this behaviour:

```ts
getError('something'); // `includeUntounched` defaults to false
getError('something', { includeUntouched: true });
```